app/lib/three_scale/spam_protection/protector.rb

```diff
@@ -60,11 +60,11 @@ def captcha_required?
       end
 
       def captcha_needed?
-        captcha_required? || (!http_method.get? && spam?)
+        captcha_required? || spam_post_request? || spam_get_request?
```

Comment:
So in the end it does not matter if it's `http_method.get?` or `!http_method.get?`? It looks like this can be reduced to 

```suggestion
        captcha_required? || considered_as_spam? || was_considered_as_spam?
```

though I'd suggest for readability

```suggestion
        captcha_required? || considered_spam?
```

and then

```ruby
def considered_spam?
  considered_as_spam? || was_considered_as_spam?
end

Reply:
I get it now. We cannot invoke both!

Reply:
Maybe just wrap `spam_post_request? || spam_get_request?` into a method then:

```suggestion
        captcha_required? || spam?
```

```ruby
def spam?
  spam_post_request? || spam_get_request?
end
```